DatadogSessionReplay/Sources/Recorder/RecordingTrigger.swift

```diff
@@ -0,0 +1,62 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import DatadogInternal
+import UIKit
+
+/// Orchestrates the process of triggering next snapshot recording.
+/// It has 2 recording triggers: `layoutSubviews` and touch event, which are achieved by using swizzling.
+internal class RecordingTrigger: UIViewHandler, UIEventHandler {
+    private var uiViewSwizzler: UIViewSwizzler? = nil
+    private var uiApplicationSwizzler: UIApplicationSwizzler? = nil
+
+    private weak var recordingCoordinator: RecordingCoordinating?
+
+    init(
+        recordingCoordinator: RecordingCoordinating,
+        shouldStartWatchingTriggers: Bool
+    ) throws {
+        self.recordingCoordinator = recordingCoordinator
+
+        uiViewSwizzler = try UIViewSwizzler(handler: self)
+        uiApplicationSwizzler = try UIApplicationSwizzler(handler: self)
+
+        if shouldStartWatchingTriggers {
+            startWatchingTriggers()
+        }
+    }
+
+    deinit {
+        uiViewSwizzler?.unswizzle()
+        uiApplicationSwizzler?.unswizzle()
+    }
+
+    func startWatchingTriggers() {
+        if recordingCoordinator?.startRecording() == true {
+            uiViewSwizzler?.swizzle()
+            uiApplicationSwizzler?.swizzle()
+        }
```

Comment:
question/ Is is this check somehow evaluated after RUM context is changed to "new session"? The underlying return value from `startRecording()` is:
```swift
return recordingEnabled && isSampled
```
where `isSampled` is updated in `onRUMContextChanged(rumContext:)`. It seems that `RecordingTrigger` doesn't re-evaluate swizzling on context change. Ultimately, this may lead to "no recording" when first session was not sampled so swizzlings are not installed, but the succeeding session is sampled but swizzlings remain to be off.

Reply:
Yeah - that's a good point. Trigger will keep firing even the next session is sampled out.